src/models/AbstractCatalogProfile/AbstractCatalogProfile.ts

```diff
@@ -295,22 +295,28 @@ export abstract class AbstractCatalogProfileStore {
     @action setColumnFilter = (filter: string, columnName: string) => {
         const current = this.catalogControlHeader.get(columnName);
         const newHeader: ControlHeader = {
-            columnIndex: current.columnIndex,
-            dataIndex: current.dataIndex,
-            display: current.display,
+            columnIndex: current?.columnIndex ?? NaN,
+            dataIndex: current?.dataIndex ?? NaN,
+            display: current?.display ?? false,
             filter: filter,
-            columnWidth: current.columnWidth
+            columnWidth: current?.columnWidth ?? null
         };
         this.catalogControlHeader.set(columnName, newHeader);
         this.updateTableStatus(true);
     };
 
     @action.bound setTableColumnWidth(width: number, columnName: string) {
-        this.catalogControlHeader.get(columnName).columnWidth = width;
+        const header = this.catalogControlHeader.get(columnName);
+        if (header) {
+            header.columnWidth = width;
+        }
     }
 
     @action setHeaderDisplay(val: boolean, columnName: string) {
-        this.catalogControlHeader.get(columnName).display = val;
+        const header = this.catalogControlHeader.get(columnName);
```

Comment:
Same comment as above. If the key is present by the value is undefined, then the has() function should not be used.

Reply:
Same comment.